src/pgmoneta_ext/lib.c

```diff
@@ -131,23 +131,16 @@ pgmoneta_ext_checkpoint(PG_FUNCTION_ARGS)
    Oid roleid;
    TupleDesc tupdesc;
    bool nulls[2];
-   bool is_superuser;
-   bool is_pg_checkpoint;
    char cp[1024];
+   int privileges;
 
    memset(nulls, 0, sizeof(nulls));
    memset(&cp, 0, sizeof(cp));
-   is_superuser = false;
-   is_pg_checkpoint = false;
 
    roleid = GetUserId();
-   is_superuser = pgmoneta_ext_check_superuser(roleid);
+   privileges = pgmoneta_ext_check_privilege(roleid);
 
-#if PG_VERSION_NUM >= 150000
-   is_pg_checkpoint = pgmoneta_ext_check_role(roleid, "pg_checkpoint");
-#endif
-
-   if (is_superuser || is_pg_checkpoint)
+   if (privileges & PRIVILEDGE_PG_CHECKPOINT || privileges & PRIVILEDGE_SUPERUSER)
```

Comment:
Perhaps it's better to use `privileges & (PRIVILEDGE_PG_CHECKPOINT |  PRIVILEDGE_SUPERUSER)`, to keep a bitmask style.